ui/pages/desktop-error/render-desktop-error.js

```diff
@@ -64,7 +72,23 @@ export function renderDesktopError({
   const renderCTA = (id, text, onClick) => {
     return (
       <Box marginTop={6}>
-        <Button type="primary" onClick={onClick ?? noop} id={id}>
+        <Button
+          type="primary"
+          onClick={() => {
+            if (onClick) {
+              onClick();
+              if (typeof trackEvent === 'function') {
+                trackEvent({
+                  category: EVENT.CATEGORIES.DESKTOP,
+                  event: `${text} Button Clicked`,
+                });
+              }
```

Comment:
These types of dynamic event names are *not* acceptable....They end up creating variadic events that make it very hard to derive meaning from. @worldlyjohn will advise you on if we want this event or not and if so how to add properties to cover the use cases.

Reply:
One quick update on this case. I pushed some changes so that instead of using the text it uses the id of the button to determine the event name. In case, more properties need to be added it's possible to extend this function to also include specific properties for each event.

Reply:
id also not sufficient as it's easy for an element id to change and unintended consequence is the event is now changed too.  usually when you want to do this, the best practice is a more generic event name and using this value as a property (though the id field should probably be validated that it matches one element in a predefined set)